### PR TITLE
Show changelog in apps management

### LINKED ISSUE #22436: Feature Req: Show change logs for app updates
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
This feature request is not related to a problem. This is a Quality Of Life feature request.

**Describe the solution you'd like**
When updating apps in the built in app store browser, it would be very nice to be able to easily see the latest change log for that app before updating. It would be great to see this at the bottom of the right side menu that appears when clicking an app for more information.

**Describe alternatives you've considered**
Another solution to this may be a change log button on the app or in the right side menu to open a popup window containing the change log for the app. Or, another solution could be a dedicated change log link next to the appstore link.

**Additional context**
Currently, before updating an app on my Nextcloud instance, I have to track down the change log for that app by following links to the app's website, the app's Github or the appstore for apps that don't provide that information... Sometimes a changelog can be very hard to find or they are not provided at all.

### PR DESCRIPTION
**Pull request by juliusknorr**
This will use the changelog provided by the most recent app store release and show it in the apps management sidebar if available.

Besides that I also did some code cleanup, migrated to use the new sidebar tabs for description/changelog and removed the list transition which had quite a negative performance impact when searching and switching between app categories.

With a changelog:
image
image

Without a changelog:
image

### CODE REVIEW

**Comment by jancborchardt** [comment]
Very nice! Only minor thing is that we should call it "What’s new" instead of "Changelog" which is a bit bland and technical. :)

**Comment by AlexOnFOSS** [comment]
Thats nice! Might it be possible to add a timestamp when the last modification of that update (changelog) has taken place?

**Review by GretaD — APPROVED** [review]
tested and works

**Review by jancborchardt — CHANGES_REQUESTED** [review]
See comment for wording change (I would add it directly but somehow often the build process for the apps in core messes stuff up for me).

**Comment by rullzer** [comment]
needs a rebase and the comment from @jancborchardt addressed

**Comment by juliushaertl** [comment]
I'll wait for https://github.com/nextcloud/server/pull/18969 with the rebase since I forgot that @skjnldsv was already worked on the sidebar migration. Once that is done I'll just pick the relevant changelog changes.

**Comment by skjnldsv** [comment]
> I'll wait for #18969 with the rebase since I forgot that @skjnldsv was already worked on the sidebar migration. Once that is done I'll just pick the relevant changelog changes.

Thanks! :bow: :heart:

**Comment by jancborchardt** [comment]
Btw @juliushaertl to make it simpler and also work better on mobile, we could just put the changelog below the description/links instead of using tabs.

This is also how the Android Play store and the iOS app store essentially do it. To alleviate the case of a very long description, we can set some max-height which will expand on click – exactly like we do for the rich workspaces.

**Comment by rullzer** [comment]
@juliushaertl still something we can tackle for 20?

**Comment by juliushaertl** [comment]
Not before https://github.com/nextcloud/server/pull/18969

**Comment by skjnldsv** [comment]
> Not before #18969

will dive into it again soon

**Comment by rullzer** [comment]
soooo?

**Comment by juliushaertl** [comment]
Force-pushed the changelog only changes now based on master.

File: apps/settings/src/components/Markdown.vue [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+export default {
+	name: 'Markdown',
+	props: {
+		text: {
+			type: String,
+			default: '',
+		},
+	},
+	computed: {
+		renderMarkdown() {
```
Comment (skjnldsv) [inline apps/settings/src/components/Markdown.vue:40]: Would be nice as a library :wink:

File: apps/settings/src/views/Apps.vue [hunk 2]
```diff
@@ -119,7 +119,23 @@
...
-			<AppDetails :app="app" />
+
+			<AppSidebarTab id="desc"
+				icon="icon-category-office"
+				:name="t('settings', 'Details')"
+				:order="0">
+				<AppDetails :app="app" />
+			</AppSidebarTab>
+			<AppSidebarTab v-if="app.appstoreData && app.releases[0].translations.en.changelog"
+				id="desca"
```
Comment (skjnldsv) [inline apps/settings/src/views/Apps.vue:130]: desca ?
```suggestion
				id="changelog"
```

**Review by skjnldsv — APPROVED** [review]
Minor nitpick! :)
Great job!!! :rocket: :heart_eyes:

File: apps/settings/src/components/Markdown.vue (hunk 1 above)
Comment (juliushaertl) [reply apps/settings/src/components/Markdown.vue:40]: Something for `@nextcloud/vue` or as a separate one?

Comment (skjnldsv) [reply apps/settings/src/components/Markdown.vue:40]: Maybe a separate one with the markdown editor?

Comment (juliushaertl) [reply apps/settings/src/components/Markdown.vue:40]: Then let's track this in https://github.com/juliushaertl/nextcloud-text/issues/207 for now

**Pull request merged**